src/calculator.erl

```diff
@@ -0,0 +1,36 @@
+-module(calculator).
+-export([add/2, subtract/2, multiply/2, divide/2, unused_function/0, main/0]).
+
+% Function to add two integers
+-spec add(integer(), integer()) -> integer().
+add(X, Y) ->
+  X + Y + 0.5. % Returns a float instead of an integer
+
+% Function to subtract two numbers
+-spec subtract(number(), number()) -> number().
+subtract(X, Y) ->
+  X - Y.
+
+% Function to multiply two numbers
+-spec multiply(number(), number()) -> number().
```

Comment:
calculator.erl:15: Invalid type specification for function calculator:multiply/2. The success typing is 
          ([any()], _) -> any()